### PR TITLE
CU-86a1xyqam - NEON3 - SEND Tokens - Implement token and amount selection

### PR DESCRIPTION
**Pull request by ricardoak25**

### CODE REVIEW

**Comment by lllwvlvwlll** [comment]
Task linked: [CU-86a1xyqam NEON3 - SEND Tokens - Implement token and amount selection](https://app.clickup.com/t/86a1xyqam)

**Comment by ricardoak25** [comment]
It's necessary to first perform the CR and merge for:
https://github.com/CityOfZion/neon-wallet/pull/2607

File: tailwind.config.js [hunk 1]
```diff
@@ -18,6 +18,9 @@ module.exports = {
       purple: {
         DEFAULT: '#9747FF',
       },
+      darkMagenta: {
+        DEFAULT: '#7D4B93',
+      },
```
Comment (raulduartep) [inline tailwind.config.js]: Enter this color inside the magenta object as 600, 700 or 800
```suggestion
      magenta: {
        700: '#7D4B93',
      },
```

File: src/renderer/src/routes/pages/Send/SelectAccount.tsx [hunk 2]
```diff
@@ -1,25 +1,33 @@
...
 import { TbChevronRight, TbStepOut } from 'react-icons/tb'
 import { IAccountState } from '@renderer/@types/store'
 import { Separator } from '@renderer/components/Separator'
+import { StyleHelper } from '@renderer/helpers/StyleHelper'
 import { useModalNavigate } from '@renderer/hooks/useModalRouter'
 
-export const SelectAccount = () => {
+type TAccountParams = {
+  selectedAccount?: IAccountState
+  handleSelectAccount: (contact: IAccountState) => void
```
Comment (raulduartep) [inline src/renderer/src/routes/pages/Send/SelectAccount.tsx]: Rename it to onSelectAccount

File: src/renderer/src/routes/modals/SelectToken/index.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        <h2 className="text-sm text-left w-full pl-[0.2em]">{t('yourBalances')}</h2>
+        <main className="w-full overflow-y-auto flex-grow my-3 flex flex-col basis-0">
+          <TokenBalanceList balanceExchange={balanceExchange} onTokenSelected={setSelectedToken} />
+        </main>
+
+        <Button
+          className="w-[16rem]"
+          type="submit"
+          label={t('selectToken')}
+          disabled={selectedToken ? false : true}
```
Comment (raulduartep) [inline src/renderer/src/routes/modals/SelectToken/index.tsx]: You can simplify it to disabled={!selectedToken}

File: src/renderer/src/routes/pages/Send/SelectToken.tsx [hunk 4]
```diff
@@ -2,22 +2,65 @@ import { Fragment } from 'react'
...
 import { Separator } from '@renderer/components/Separator'
+import { StringHelper } from '@renderer/helpers/StringHelper'
+import { StyleHelper } from '@renderer/helpers/StyleHelper'
+import { useModalNavigate } from '@renderer/hooks/useModalRouter'
 
-export const SelectToken = () => {
+type TTokenParams = {
+  selectedAccount?: IAccountState
+  selectedToken?: TokenBalance | null
+  handleSelectToken?: (token: TokenBalance) => void
```
Comment (raulduartep) [inline src/renderer/src/routes/pages/Send/SelectToken.tsx]: Rename it to onSelectToken

File: src/renderer/src/routes/pages/Send/index.tsx [hunk 5]
```diff
@@ -14,9 +17,46 @@ import { TokenBalanceList } from './TokenBalanceList'
 import { TotalFee } from './TotalFee'
 
 export const SendPage = () => {
+  enum SendPageStep {
+    SelectAccount = 1,
+    SelectToken = 2,
+    SelectAmount = 3,
+    SelectContact = 4,
+  }
```
Comment (raulduartep) [inline src/renderer/src/routes/pages/Send/index.tsx]: Move this out of the component

File: src/renderer/src/routes/pages/Send/index.tsx [hunk 6]
```diff
@@ -14,9 +17,46 @@ import { TokenBalanceList } from './TokenBalanceList'
...
+
   const { t } = useTranslation('pages', { keyPrefix: 'send' })
   const { accounts } = useAccountsSelector()
-  const balanceExchange = useBalancesAndExchange(accounts)
+  const balanceExchange = useBalancesAndExchange(accounts[0])
+  const [selectedAccount, setSelectedAccount] = useState<IAccountState>()
+  const [selectedToken, setSelectedToken] = useState<TokenBalance | null>()
+  const [selectedAmount, setSelectedAmount] = useState<number>(0)
+  const [currentStep, setCurrentStep] = useState<number>(SendPageStep.SelectAccount)
```
Comment (raulduartep) [inline src/renderer/src/routes/pages/Send/index.tsx]: Use the enum to type this state
```suggestion
  const [currentStep, setCurrentStep] = useState<SendPageStep>(SendPageStep.SelectAccount)
```

File: src/renderer/src/routes/pages/Send/SendAmount.tsx [hunk 7]
```diff
@@ -2,25 +2,85 @@ import { Fragment } from 'react'
...
+import { StyleHelper } from '@renderer/helpers/StyleHelper'
+import { useBalancesAndExchange } from '@renderer/hooks/useBalancesAndExchange'
+import { useModalNavigate } from '@renderer/hooks/useModalRouter'
 
-export const SendAmount = () => {
+type TAmountParams = {
+  selectedAccount?: IAccountState | null
+  selectedToken?: TokenBalance | null
+  selectedAmount: number
+  handleSelectAmount?: (amount: number) => void
```
Comment (raulduartep) [inline src/renderer/src/routes/pages/Send/SendAmount.tsx]: Rename it to onSelectAmount

File: src/renderer/src/routes/pages/Send/SendAmount.tsx [hunk 8]
```diff
@@ -2,25 +2,85 @@ import { Fragment } from 'react'
...
+}: TAmountParams) => {
   const { t } = useTranslation('pages', { keyPrefix: 'send' })
+  const { modalNavigateWrapper } = useModalNavigate()
+  const balanceExchange = useBalancesAndExchange(selectedAccount ? [selectedAccount] : [])
+
+  const calculateEstimatedFiatValue = (amount: number) => {
+    if (selectedToken) {
+      const totalEstimatedValue =
+        BalanceHelper.convertBalanceToCurrency(selectedToken, balanceExchange.exchange.data)?.convertedAmount ?? 0
+      const pricePerToken = totalEstimatedValue / selectedToken.amountNumber
```
Comment (raulduartep) [inline src/renderer/src/routes/pages/Send/SendAmount.tsx]: You can directly use the balanceExchange to get the price per token.
```suggestion
      const pricePerToken = BalanceHelper.getExchangeRatio(
        selectedToken.token.hash,
        selectedToken.blockchain,
        balanceExchange.exchange.data
      )

      return FilterHelper.currency(amount * pricePerToken)
```

**Pull request merged**